javascripts/discourse/components/experimental-screen.gjs

```diff
@@ -0,0 +1,56 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import { htmlSafe } from "@ember/template";
+
+export default class ExperimentalScreen extends Component {
+  @tracked left = 0;
+  @tracked right = 0;
+
+  willDestroy() {
+    super.willDestroy(...arguments);
+    window.removeEventListener("resize", this.calculateDistance);
+  }
+
+  getDistance(element) {
+    const rect = element.getBoundingClientRect();
+    return rect;
+  }
+
+  calculateDistance() {
```

Comment:
Probably worth throttling this...you can do something like this:

```javascript
calculateDistance() {
  this._throttledCalculateDistanceHandler = throttle(this, this._throttledCalculateDistance, 100);
}

_throttledCalculateDistance() {
  // the code originally in the calculateDistance function
}
```

and then in `willDestroy` you just need:

```
cancel(this._throttledCalculateDistanceHandler)
```

From `import { cancel, throttle } from "@ember/runloop";`

Reply:
Maybe use `INPUT_DELAY` from `import { INPUT_DELAY } from "discourse/lib/environment";` instead of `100` tho